### DISCUSSION TITLE
CAN WE USE EBSD DATA (IN .ANG format) to create a 2D MICROSTRUCTURE FOR ABAQUS SIMULATION ?

### DISCUSSION THREAD

**Discussion by Abhinab-iit-112** [neperfepx/neper#953] [Q&A] [answered]
Dear sir,
I have made a 2D model using neper and i also generated the (.tesr file) but the data is in binary16 not in ASCII format.
![n34p_random](https://github.com/user-attachments/assets/2d9178f0-f6a1-421c-a9c4-7e6b8bb0e424)
![n34p_random_mesh](https://github.com/user-attachments/assets/0538c328-64dc-4c16-92fb-587fa4ed5068)
(MY .tesr file: - ***tesr
 **format
   2.1
 **general
   2
   28 57
   0.035714285714 0.035087719298
 **cell
   34
  *id
   1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27
28 29 30 31 32 33 34
  *ori
   rodrigues:active
  -0.183774804250    0.111905913278    0.281901881831
  -0.299912616678    0.124959486824   -0.004604856510
  -0.090787587942    0.222897846624   -0.095418211151
   0.083904117527   -0.248033306507   -0.403335242202
  -0.398377071322    0.112977113614    0.376405627550
  -0.094805010944   -0.120208721692   -0.359323627483
  -0.243485319271    0.337477886104    0.361455267220
  -0.108185936383   -0.219710152869    0.291711931452
   0.151752194310    0.058946530336    0.278620667740
  -0.042253779318   -0.381825216746   -0.206135985603
   0.012419450539    0.079646594685   -0.136232665976
  -0.245234384307   -0.052841185792   -0.350965073114
   0.038193472944    0.392815661804    0.082707272642
  -0.384781572485   -0.141428834488   -0.211330207565
  -0.307439887917   -0.126281454273   -0.316137887588
   0.326348734314    0.064939201877    0.123678880335
   0.140323005801    0.053131487136    0.267428959425
  -0.019634514268   -0.031642267356   -0.167096972780
   0.040186448258    0.413036302338   -0.111987248989
   0.124742160709    0.171798711593    0.075227427651
  -0.245744522613   -0.286491946043   -0.094168931843
  -0.378867097494    0.190284508791   -0.036010965673
   0.251972103924   -0.245265433167   -0.316366033866
  -0.133625741468    0.324051624862    0.321557117380
   0.070168225309   -0.121168433081   -0.127303654820
   0.025557810295   -0.400417847857   -0.347566564285
   0.162966466625   -0.065203019264    0.223566608268
   0.259245705052    0.102073143606   -0.030448606451
  -0.183326925022    0.163949653064   -0.202486149591
   0.130498082239    0.399448459567   -0.079265739527
  -0.236220875562   -0.324159069080    0.068018945949
  -0.053936103553   -0.338224899001    0.220170842397
   0.354352507527    0.406172105103   -0.143411897841
  -0.342851502344   -0.317755514725    0.148337199581
  *crysym
   cubic
 **data
   binary16
                                                                                                                                                                                                                                                                                                           
 
                          
 
 
        	 	 	                
 
 
 
       	 	 	 	 	             
 
 
 
 
 
      	 	 	 	 	 	 	          
 
 
 
 
 
 
      	 	 	 	 	 	 	 	 	 	  " "         
 
 
 
 
      	 	 	 	 	 	 	 	 	 " " " " "                      	 	 	 	 	 	 	 	 " " " " " "                      	 	 	 	 	 	 	 " " " " " " "                     	 	 	 	 	 	 " " " " " "                        	 	 	 " " " " " "                         	 " " " " "                         "                                                                                                                                                             ! ! ! ! !                      ! ! ! ! ! ! ! !                    ! ! ! ! ! ! ! ! !                  ! ! ! ! ! ! ! ! ! ! !                   ! ! ! ! ! ! ! ! ! !                    ! ! ! ! ! ! ! ! !                     ! ! ! ! ! ! ! !                       ! ! ! ! ! !     
                   ! ! ! ! !   
 
 
 
                         
 
 
 
                         
 
 
 
 
                        
 
 
 
 
                         
 
 
 
 
                        
 
 
 
 
                        
 
 
 
 
 
                       
 
 
 
 
 
                       
 
 
 
 
 
 
                       
 
 
 
 
 
                       
 
 
 
 
                        
 
 
                                                                                                                                                                                                                                                                                  
***end
)
PLEASE provide me the detailed step to generate the microstructure(2D model i attached) using EBSD data.
please help me with the CODE TO GENERATE SUCH STRUCTURE.
@rquey 
Waiting for your Positive response.

**Comment by rquey** [comment, marked as answer]
Use "[-tesrformat](https://neper.info/doc/neper_t.html#cmdoption-tesrformat) `ascii`".

**Comment by Abhinab-iit-112** [comment]
![img2024](https://github.com/user-attachments/assets/3e1a6060-17e2-4d6d-bf35-422e29f7c5ff)
Dear sir,
i have created it
but unable to create the INP files for ABAQUS.
(abhinabrath1998@ABHINAB-IIT01:/mnt/c/Ubuntu_software_app$ neper -M n34_random.tesr -dim 2 \
        -elttype tet \
        -rcl 1 \
        -order 2 \
        -o trial34 \
        -format msh,inp

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-2
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 4 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/abhinabrath1998/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n34_random.tesr -dim 2 -elttype tet -rcl 1 -order 2
         -o trial34 -format msh,inp
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading raster tessellation...
Info   :     [i] Parsing file `n34_random.tesr'...
Info   :     [i] Parsed file `n34_random.tesr'.
Info   :   - Reconstructing topology...
Info   :     > Reconstructing 1D mesh... 100%
Info   :     > Reconstructing 0D mesh... 100%
Error  : You have discovered a bug in Neper!  Please file an issue at
         https://github.com/rquey/neper/issues.  Thank you.

Aborted (core dumped))

can you help me in a code that can create INP file and also visualize the MESH
@rquey 
please help me in this

**Comment by rquey** [comment]
You have to do some cleaning to resolve topology issues. See [`-transform rmsat,grow`](https://neper.info/doc/neper_t.html#cmdoption-transform).

**Comment by Abhinab-iit-112** [comment]
Dear sir,
**if you can provide with a full code like to do cleaning and then doing meshing and then creating INP files for ABAQUS simulation and also visualizing the meshes 
i will be highly grateful sir**
i am unable to get (You have to do some cleaning to resolve topology issues. See [-transform rmsat,grow](https://neper.info/doc/neper_t.html#cmdoption-transform).)
please sir if you can give me code framework how to clean and create INP files for Abaqus simulation.


**ONE MORE THING SIR
IN tesr file in Rodrigues : active
the orientation angles i can insert of my own obtained from EBSD data ???
and all other things keeping same like cleaning and all, i have to make the INP files ?**

@rquey sir looking for you positive response
please sir help me in framing those CODES mentioned above.
